### PR TITLE
Replace boost::string_ref with std::string_view

### PR DESCRIPTION
**Pull request by DennisOSRM**
# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

Please read our [documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/releasing.md) on release and version management.
If your PR is still work in progress please attach the relevant label.

## Tasklist

 - [x] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [x] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [x] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: include/util/guidance/name_announcements.hpp [hunk 1]
```diff
@@ -27,8 +27,8 @@ namespace guidance
 // Name Change Logic
 // Used both during Extraction as well as during Post-Processing
 
-inline util::StringView longest_common_substring(const util::StringView &lhs,
-                                                 const util::StringView &rhs)
+inline std::string_view longest_common_substring(const std::string_view &lhs,
```
Comment (SiarheiFedartsou) [inline include/util/guidance/name_announcements.hpp:30]: nit. it seems that it is better to pass `string_view` by reference https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/

We had reference here already, but as we are changing it anyway...

File: include/util/guidance/name_announcements.hpp [hunk 2]
```diff
@@ -70,7 +70,9 @@ template <typename StringView> inline auto decompose(const StringView &lhs, cons
     const auto trim = [](StringView view) {
         // we compare suffixes based on this value, it might break UTF chars, but as long as we are
         // consistent in handling, we do not create bad results
-        std::string str = boost::to_lower_copy(view.to_string());
+        std::string str(view);
```
Comment (SiarheiFedartsou) [inline include/util/guidance/name_announcements.hpp:73]: nit.

seems to be a little bit faster to do smth like:
```
std::string str;
str.reserve(view.size());
for (auto c : view) {
  str += std::tolower(ch);
}
```
(it probably can be done the same way with some STL algorithms too)

**Review by SiarheiFedartsou — APPROVED** [review]

**Comment by SiarheiFedartsou** [comment]
Nice to see that you are actively contributing again 🚀

**Comment by SiarheiFedartsou** [comment]
JFYI there is some annoying issue with Conan installation on macOS executors on CI - I am trying to fix it now. Workaround is to use `brew install conan` instead of `python3 -m pip install conan`(but it is not recommended as I can see in Conan docs).

**Comment by DennisOSRM** [comment]
> JFYI there is some annoying issue with Conan installation on macOS executors on CI - I am trying to fix it now. Workaround is to use `brew install conan` instead of `python3 -m pip install conan`(but it is not recommended as I can see in Conan docs).

Thanks for letting me know. I'll rebase the PR once the bug is fixed, then.

File: include/util/guidance/name_announcements.hpp [hunk 3]
```diff
@@ -70,7 +70,10 @@ template <typename StringView> inline auto decompose(const StringView &lhs, cons
     const auto trim = [](StringView view) {
         // we compare suffixes based on this value, it might break UTF chars, but as long as we are
         // consistent in handling, we do not create bad results
-        std::string str = boost::to_lower_copy(view.to_string());
+        std::string str;
+        str.reserve(view.size());
+        std::transform(
+            str.begin(), str.end(), str.begin(), [](unsigned char c) { return std::tolower(c); });
```
Comment (SiarheiFedartsou) [inline include/util/guidance/name_announcements.hpp:76]: I think you have to use std::back_inserter here…

**Pull request merged**